src/sage/combinat/combinat.py

```diff
@@ -1174,11 +1173,11 @@ def __init__(self, l, copy=True):
 
         Test indirectly that we copy the input (see :trac:`18184`)::
 
-            sage: L = IntegerListsLex(element_class=Partition)
-            sage: x = [3, 2, 1]
-            sage: P = L(x)
-            sage: x[0] = 5
-            sage: list(P)
+            sage: L = IntegerListsLex(element_class=Partition)                          # optional - sage.combinat
```

Comment:
It's surprising to add optional tag `sage.combinat` inside files from the combinat module. If it is necessary to add this tag, we should add it directly at the top of all files of the combinat module.

Reply:
I've started to document the rules for this in #35749 -- see https://deploy-preview-35749--sagemath-tobias.netlify.app/reference/misc/sage/features/sagemath.html#sage.features.sagemath.sage__combinat

In short, some of the most basic modules in `sage.combinat` are needed already at a very low level. `sage.combinat.combinat`, `.permutations`, `combinations`, etc. will therefore be shipped by the distribution sagemath-categories (see https://github.com/sagemath/sage/blob/7992ac088e1f73ca3dd4db9fc6560a704cd5034b/pkgs/sagemath-categories/MANIFEST.in.m4#L180 - in #35095).
These modules then need to mark uses of higher-level modules from `sage.combinat` with `# optional - sage.combinat`. (Here it is the use of `Partitions` that makes it necessary.)